oop2-encapsulate-inherit.js

```diff
@@ -5,27 +5,269 @@
+            this.#user = user;
+        }
+    }
+
+
+
+
+
+
+/* 2. Create the following classes:
     - Umbrella class.  It should have a an organization name.  It should only have a single instance.  Umbrella is the "Parent Organization" e.g. Allegis Group
     - Company class. It should have a compnay name and # of employees.  Create at least 3 different companies that are children of the Umbrella Organization.  (e.g. TEKsystems, Aerotek, etc.)
     - Site class. It should have the name of the company, and the location of the site.  It is a child class that inherits from the Company class.  Create 3 sites for each company.
     - Employee class. It should have a employee name, job title, and salary properties. Create 10 employee instances: 1 CEO, 1 Manager, 1 Secretary, 2 Engineers, 1 Financial Officer, 1 Janitor, 1 Marketer, 1 HR Personnel, 1 Lawyer.  Feel free to add other employee roles at your own discretion.
     
     - For each class, add 2 properties and 2 methods to each.  The properties and methods should make sense, considering what the classes are supposed to be representing.
-
+*/
 
 
     //your code here...
 
+    class Umbrella {
+        #orgName;
+        #yearFormed = 1999;
+        #listOfCompanies = ['TEKsystems', 'Aerotek', 'TEK Global'];
+        #updatedList;
+
+        constructor(orgName){
+            this.#orgName = orgName;
+        }
+
+        acquireCompany(compName){
+            if(this.#listOfCompanies.includes(compName)){
+                console.log(`${compName} is already in the Allegis Group, and cannot be acquired.`);
+            } else {
+                this.#listOfCompanies.push(compName);
+            }
+            return this.#this.#listOfCompanies;
+        }
+
+        consolidateCompany(compName){
+            if(this.#listOfCompanies.includes(compName)){
+                this.#updatedList = this.#listOfCompanies.filter(function(company) {return company !== compName});
+            } else{
+                console.log(`${compName} is not in the Alegis group, and cannot be consolidated.`);
+            }
+            return this.#updatedList;
+        }
+    }
+
+    class Company extends Umbrella {
+        #compName;
+        #employeeCount;
+        #compMotto;
+        #yearFounded;
+
+        constructor(compName, employeeCount, compMotto, yearFounded){
+            super();
+            this.#compName = compName;
+            this.#compMotto = compMotto;
+            this.#yearFounded = yearFounded;
+        }
+
+        mottoChange(newMotto){
+            return this.#compMotto = newMotto;
+        }
+
+        tellNews(){
+            console.log(`${this.#compName} is catering lunch for all ${this.#employeeCount} employees in the company tomorrow!`);
+        }
+    }
+
+    class Site extends Company {
+        #location;
+        #compName;
+        #inBusiness = true;
+        #hasRemote = true;
+
+        constructor(compName, location){
+            super(comnpName);
```

Comment:
typo...you can't have typos in pull requests. A pull request says "My work is done and perfect. This is ready to go in front of the client." Typos in code means it doesn't work